### DISCUSSION TITLE
질문드려요

### DISCUSSION THREAD

**Discussion by uyoung9** [yoonjeong-kwon/fastcampus-kubernetes#6] [Q&A] [answered]
ch4.02 service clusterIP 서비로 pod 노출하는 방법에서 실습2 

실습2에서 환경변수로 접속 부분에서

환경변수는 쿠버네티스가 자동으로 생성해주는 건지, 아니면 직접 환경변수를 미리 입력해둔건지 궁금합니다. 

감사합니다.

**Comment by yoonjeong-kwon** [comment, marked as answer]
@uyoung9 님, 안녕하세요.

말씀하신 환경변수는 쿠버네티스가 파드 컨테이너 안에 설정한 값 입니다.
쿠버네티스가 파드를 생성하는 시점에 클러스터 내에 존재하는 Service 오브젝트들에 대한 환경변수를 파드 컨테이너에 설정해줍니다. 
(참고: Ch 04. 쿠버네티스로 서비스 운영하기 - 01. Service 소개의 슬라이드 9번)

예를 들어 A라는 파드를 생성할 때, AA라는 Service가 이미 존재한다면 쿠버네티스는 AA_SERVICE_HOST, AA_SERVICE_PORT와 같은 환경변수를 파드 컨테이너 내에 설정합니다. 사용자가 지정한 Service 이름 기반으로 정해진 이름 규칙에 따라 Service 관련 환경변수를 설정해주기 때문에 애플리케이션 개발 시 환경변수를 이용해 다른 Pod와 통신하도록 만들 수도 있습니다.

아래는 payment 서비스를 생성한 후, order 파드를 생성하여 컨테이너 접속 후 출력한 환경변수 목록입니다.
(참고: Ch 04. 쿠버네티스로 서비스 운영하기 - 02. [실습] Service - ClusterIP 서비스로 Pod 노출하는 방법의 슬라이드 9번)

```bash
$ kubectl exec order-7954c88f5b-kqnf4 -n snackbar -- env | grep PAYMENT
PAYMENT_PORT_80_TCP_PORT=80
PAYMENT_SERVICE_HOST=10.80.5.16
PAYMENT_SERVICE_PORT=80
PAYMENT_PORT_80_TCP_PROTO=tcp
PAYMENT_PORT=tcp://10.80.5.16:80
PAYMENT_PORT_80_TCP=tcp://10.80.5.16:80
PAYMENT_PORT_80_TCP_ADDR=10.80.5.16
````
감사합니다.
